packages/components/src/Grid.astro

```diff
@@ -0,0 +1,219 @@
+---
+const { container, fluid, xs, sm, md, lg, auto, gap } = Astro.props;
+const containerClass = container ? "grid-container" : "";
+const autoClass = auto ? "auto-columns" : "";
+const gapClass = gap ? `gap-${gap}` : "";
+const xsClass = xs ? `col-${xs}` : "";
+const smClass = sm ? `col-sm-${sm}` : "";
+const mdClass = md ? `col-md-${md}` : "";
+const lgClass = lg ? `col-lg-${lg}` : "";
+const fluidClass = fluid ? "fluid" : "";
+const classList = [
+  containerClass,
+  fluidClass,
+  autoClass,
+  gapClass,
+  xsClass,
+  smClass,
+  mdClass,
+  lgClass,
+]
+  .join(" ")
+  .trim();
+---
+
+<style lang="scss">
+  .grid-container {
+    display: grid;
+    grid-template-columns: repeat(12, 1fr);
+    gap: 16px;
+    max-width: 1440px;
+    margin: 0 auto;
+    padding: 0 16px;
+
+    &.fluid {
+      max-width: 100%;
+    }
+
+    @media (min-width: 768px) {
+      padding: 0 32px;
+    }
+  }
+
+  .auto-columns {
+    grid-template-columns: repeat(auto-fill, minmax(min-content, 1fr));
+  }
+
+  .gap-1 {
+    gap: 4px;
+  }
+  .gap-2 {
+    gap: 8px;
+  }
+  .gap-3 {
+    gap: 16px;
+  }
+  .gap-4 {
+    gap: 24px;
+  }
+  .gap-5 {
+    gap: 32px;
+  }
```

Comment:
Not necessary to address in this PR but I wonder if there's a way to also accept a generic value for gap like:

```astro
<Grid gap="40px" />
```

I'm not sure how to do this with Astro tbh